drivers/npmx/npmx_shell.c

```diff
@@ -1649,6 +1658,330 @@ static int cmd_ldsw_ldo_voltage_get(const struct shell *shell, size_t argc, char
 	return 0;
 }
 
+npmx_gpio_mode_t gpio_mode_convert(uint8_t mode)
+{
+	switch (mode) {
+	case 0:
+		return NPMX_GPIO_MODE_INPUT;
+	case 1:
+		return NPMX_GPIO_MODE_INPUT_OVERRIDE_1;
+	case 2:
+		return NPMX_GPIO_MODE_INPUT_OVERRIDE_0;
+	case 3:
+		return NPMX_GPIO_MODE_INPUT_RISING_EDGE;
+	case 4:
+		return NPMX_GPIO_MODE_INPUT_FALLING_EDGE;
+	case 5:
+		return NPMX_GPIO_MODE_OUTPUT_IRQ;
+	case 6:
+		return NPMX_GPIO_MODE_OUTPUT_RESET;
+	case 7:
+		return NPMX_GPIO_MODE_OUTPUT_PLW;
+	case 8:
+		return NPMX_GPIO_MODE_OUTPUT_OVERRIDE_1;
+	case 9:
+		return NPMX_GPIO_MODE_OUTPUT_OVERRIDE_0;
+	default:
+		return NPMX_GPIO_MODE_INVALID;
+	}
+}
+
+npmx_gpio_pull_t gpio_pull_convert(uint8_t pull)
+{
+	switch (pull) {
+	case 0:
+		return NPMX_GPIO_PULL_DOWN;
+	case 1:
+		return NPMX_GPIO_PULL_UP;
+	case 2:
+		return NPMX_GPIO_PULL_NONE;
+	default:
+		return NPMX_GPIO_PULL_INVALID;
+	}
+}
+
+npmx_gpio_drive_t gpio_drive_convert(uint8_t drive)
+{
+	switch (drive) {
+	case 0:
+		return NPMX_GPIO_DRIVE_1_MA;
+	case 1:
+		return NPMX_GPIO_DRIVE_6_MA;
+	default:
+		return NPMX_GPIO_DRIVE_INVALID;
+	}
+}
+
+npmx_error_t gpio_config_type_set(const struct shell *shell, gpio_config_type_t config_type,
+				  npmx_gpio_config_t *gpio_config, uint32_t input_arg)
+{
+	if (config_type == GPIO_CONFIG_TYPE_MODE) {
+		gpio_config->mode = gpio_mode_convert(input_arg);
+
+		if (gpio_config->mode == NPMX_GPIO_MODE_INVALID) {
+			shell_error(shell, "Error: mode value can be:");
+			shell_error(shell, "       0 - Input");
+			shell_error(shell, "       1 - Input logic 1");
+			shell_error(shell, "       2 - Input logic 0");
+			shell_error(shell, "       3 - Input rising edge event");
+			shell_error(shell, "       4 - Input falling edge event");
+			shell_error(shell, "       5 - Output interrupt");
+			shell_error(shell, "       6 - Output reset");
+			shell_error(shell, "       7 - Output power loss warning");
+			shell_error(shell, "       8 - Output logic 1");
+			shell_error(shell, "       9 - Output logic 0");
+			return NPMX_ERROR_INVALID_PARAM;
+		}
+	} else if (config_type == GPIO_CONFIG_TYPE_PULL) {
+		gpio_config->pull = gpio_pull_convert(input_arg);
+
+		if (gpio_config->pull == NPMX_GPIO_PULL_INVALID) {
+			shell_error(
+				shell,
+				"Error: pull value can be 0-pull down, 1-pull up, 2-pull disable.");
+			return NPMX_ERROR_INVALID_PARAM;
+		}
+	} else if (config_type == GPIO_CONFIG_TYPE_DRIVE) {
+		gpio_config->drive = gpio_drive_convert(input_arg);
+
+		if (gpio_config->drive == NPMX_GPIO_DRIVE_INVALID) {
+			shell_error(shell, "Error: drive value can be 0-1mA or 1-6mA.");
```

Comment:
```suggestion
			shell_error(shell, "Error: drive value can be 0-1 mA or 1-6 mA.");
```
@annakielar  should we add a space here?

Reply:
In general the answer is yes. On the other hand, I wouldn't separate mA in this particular case because 0-1 and 1-6 notations are very confusing - they look like ranges, and not as values assigned to 0 and 1. We can also try "0 - 1 mA" with spaces before and after -. Please decide which version is better.